graphql_compiler/compiler/emit_sql.py

```diff
@@ -876,25 +906,29 @@ def unfold(self):
         self._aliases[subquery_alias_key] = fold_subquery
         self._from_clause = from_cls
 
-        # Ensure references to the outer vertex table after the Unfold refer to a totally new
-        # copy of the outer vertex table. Otherwise references would select columns from the
-        # copy of that table found inside the fold subquery.
-        self._aliases[(self._current_location.at_vertex().query_path, None)] = outer_vertex
-
         # clear the fold from the compilation state
         self._current_fold = None
         self._fold_vertex_location = None
         self._current_alias = outer_vertex
+        self._current_location = self._current_location.base_location
 
     def mark_location(self):
         """Execute a MarkLocation Block."""
-        # If the current location is the beginning of a fold, the current alias
-        # will eventually be replaced by the resulting fold subquery during Unfold.
-        self._aliases[
-            (self._fold_vertex_location.base_location.query_path,
-             self._fold_vertex_location.fold_path)
-            if self._current_fold is not None else (self._current_location.query_path, None)
-        ] = self._current_alias
+        if self._current_fold is not None:
+            # If we are in a fold, then we visit the current vertex
+            self._current_fold.visit_vertex(self._current_alias,
+                                            self._current_location,
+                                            self._all_folded_fields)
```

Comment:
You're passing `all_folded_fields` for `all_folded_outputs`, which smells buggy.